domains/client/domain-operator/src/tests.rs

```diff
@@ -3411,6 +3411,131 @@ async fn existing_bundle_can_be_resubmitted_to_new_fork() {
+    produce_blocks!(ferdie, alice, 1).await.unwrap();
+
+    // add consensus chain to domain chain allow list
+    ferdie
+        .construct_and_send_extrinsic_with(subspace_test_runtime::RuntimeCall::Messenger(
+            pallet_messenger::Call::initiate_domain_update_chain_allowlist {
+                domain_id: GENESIS_DOMAIN_ID,
+                update: ChainAllowlistUpdate::Add(ChainId::Consensus),
+            },
+        ))
+        .await
+        .expect("Failed to construct and send domain chain allowlist update");
+
+    // produce another block so allowlist on domain are updated
+    produce_blocks!(ferdie, alice, 1).await.unwrap();
+
+    // Open channel between the Consensus chain and EVM domains
+    alice
+        .construct_and_send_extrinsic(evm_domain_test_runtime::RuntimeCall::Messenger(
+            pallet_messenger::Call::initiate_channel {
+                dst_chain_id: ChainId::Consensus,
+                params: pallet_messenger::InitiateChannelParams {
+                    max_outgoing_messages: 100,
+                },
+            },
+        ))
+        .await
+        .expect("Failed to construct and send extrinsic");
+
+    // Wait until channel open
+    produce_blocks_until!(ferdie, alice, {
+        alice
+            .get_open_channel_for_chain(ChainId::Consensus)
+            .is_some()
+    })
+    .await
+    .unwrap();
+
+    produce_blocks!(ferdie, alice, 20).await.unwrap();
+
+    // close channel on Domain using domain sudo.
+    // Sudo on consensus chain will send a sudo call to domain
+    // once the call is executed in the domain,
+    // closing channel flow should happen and channel on both domain
+    // consensus chain should be closed.
+    let channel_id = alice
+        .get_open_channel_for_chain(ChainId::Consensus)
+        .unwrap();
+    let sudo_unsigned_extrinsic = alice
+        .construct_unsigned_extrinsic(evm_domain_test_runtime::RuntimeCall::Messenger(
+            pallet_messenger::Call::close_channel {
+                chain_id: ChainId::Consensus,
+                channel_id,
+            },
+        ))
+        .encode();
+    ferdie
+        .construct_and_send_extrinsic_with(pallet_sudo::Call::sudo {
+            call: Box::new(subspace_test_runtime::RuntimeCall::Domains(
+                pallet_domains::Call::send_domain_sudo_call {
+                    domain_id: GENESIS_DOMAIN_ID,
+                    call: sudo_unsigned_extrinsic,
+                },
+            )),
+        })
+        .await
+        .expect("Failed to construct and send consensus chain to close channel");
+
+    // Wait until channel close
+    produce_blocks_until!(ferdie, alice, {
+        alice
+            .get_open_channel_for_chain(ChainId::Consensus)
+            .is_none()
+    })
+    .await
+    .unwrap();
+
+    // produce 150 more blocks to ensure nothing went wrong
+    produce_blocks!(ferdie, alice, 150).await.unwrap();
```

Comment:
Why do we have to produce such many blocks? the challenge period and confirmation k were set to a small value in the test.

Reply:
I'm being more pessimistic here to ensure we can produce enough and nothing broke